packages/stargate/src/queries/auth.ts

```diff
@@ -0,0 +1,60 @@
+import { assert } from "@cosmjs/utils";
+
+import { cosmos, google } from "../generated/codecimpl";
+import { QueryClient } from "./queryclient";
+import { toAccAddress, toObject } from "./utils";
+
+export interface AuthExtension {
+  readonly auth: {
+    readonly account: (address: string) => Promise<cosmos.auth.IBaseAccount | null>;
+    readonly unverified: {
+      readonly account: (address: string) => Promise<cosmos.auth.IBaseAccount | null>;
+    };
+  };
+}
+
+export function setupAuthExtension(base: QueryClient): AuthExtension {
+  // Use this service to get easy typed access to query methods
+  // This cannot be used to for proof verification
+  const queryService = cosmos.auth.Query.create((method: any, requestData, callback) => {
+    // Parts of the path are unavailable, so we hardcode them here. See https://github.com/protobufjs/protobuf.js/issues/1229
+    const path = `/cosmos.auth.Query/${method.name}`;
+    base
+      .queryUnverified(path, requestData)
+      .then((response) => callback(null, response))
+      .catch((error) => callback(error));
+  });
+
+  return {
+    auth: {
+      account: async (address: string) => {
```

Comment:
I like this... unverified is more or less autogen, and all the other proven, handl-built ones are top-level.

Good stuff